hydra_base/db/model.py

```diff
@@ -506,111 +512,460 @@ class ResourceAttrMap(Base, Inspect):
 
 class Template(Base, Inspect):
     """
+    Template
     """
 
-    __tablename__='tTemplate'
+    __tablename__ = 'tTemplate'
 
     id = Column(Integer(), primary_key=True, nullable=False)
-    name = Column(String(200),  nullable=False, unique=True)
+    parent_id = Column(Integer(), ForeignKey('tTemplate.id'))
+    name = Column(String(200), unique=True)
     description = Column(String(1000))
-    cr_date = Column(TIMESTAMP(),  nullable=False, server_default=text(u'CURRENT_TIMESTAMP'))
-    layout  = Column(Text().with_variant(mysql.LONGTEXT, 'mysql'),  nullable=True)
+    cr_date = Column(TIMESTAMP(), nullable=False, server_default=text(u'CURRENT_TIMESTAMP'))
+    layout = Column(Text().with_variant(mysql.LONGTEXT, 'mysql'))
```

Comment:
Good question.
The template layout isn't used, but is intended to define defaults at a higher level than a type -- ex: I want all nodes in this template to be green squares instead of the global default black circle.

If you have a suggestion for a different column type for layouts, i'd be glad to hear them, but in a separate issue as this one is already packed!

The layouts can get very big for template types when storing node shapes. the GOS shape of a node can contain hundreds or thousands of points.